vagrant/catalog/teacherActions.py

```diff
@@ -0,0 +1,409 @@
+import flask
+from flask import Flask, render_template, request, redirect, url_for, jsonify
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from database_setup import Base, Teacher, Student, Goal, StudentGoalLink, Group, StudentGroupLink
+from sqlalchemy import DateTime
+app = Flask(__name__)
+
+engine = create_engine('sqlite:///teacheractions.db')
+
+Base.metadata.bind = engine
+
+DBSession = sessionmaker(bind=engine)
+session = DBSession()
+@app.route('/loggedin/<int:group_id>/showallstudentsingroup', methods=['GET'])
+def showAllStudentsInGroupjson(group_id):
+    wanted_group = session.query(Group).filter_by(id=group_id).one_or_none()
+    student_list = []
+    for student in wanted_group.students
+    student_list.append(student.serialize)
+    return flask.jsonify(student_list)
+
+
+def showAllStudentsInGroup(group_id, session):
+    wanted_group = session.query(Group).filter_by(id=group_id).one_or_none()
+    return wanted_group.students
+
+
+@app.route('/loggedin/<int:student_id>/showallstudentgoals', methods=['GET'])
+def showAllStudentGoals(student_id):
+    wanted_student = session.query(Student).filter_by(
+        id=student_id).one_or_none()
+    goal_list = []
+    for goal in wanted_student.goals
+    goal_list.append(goal.serialize)
+    return flask.jsonify(goal_list)
+
+
+def getGroupByTeacher(teacher_id, session):
```

Comment:
A teacher will be part of multiple groups so this should get all the groups associated with the teacher